interfaces/cython/cantera/test/test_composite.py

```diff
@@ -224,6 +224,14 @@ def test_collect_data(self):
         self.assertIn('X', collected)
         self.assertEqual(collected['X'].shape, (0, self.gas.n_species))
 
+    def test_getitem(self):
+        states = ct.SolutionArray(self.gas, 10, extra={"index": range(10)})
+        for ix, state in enumerate(states):
+            assert state.index == ix
+
+        assert list(states[:2].index) == [0, 1]
+        assert list(states[100:102].index) == [] # outside of range
```

Comment:
It seems like this should raise an `IndexError`, but insofar as this maintains current behavior, it seems fine to me to keep it like this and defer adding the `IndexError`.